usr/src/uts/common/fs/zfs/sys/dmu.h

```diff
@@ -102,16 +103,18 @@ typedef enum dmu_object_byteswap {
 
 #define	DMU_OT_NEWTYPE 0x80
 #define	DMU_OT_METADATA 0x40
-#define	DMU_OT_BYTESWAP_MASK 0x3f
+#define	DMU_OT_ENCRYPTED 0x20
+#define	DMU_OT_BYTESWAP_MASK 0x1f
```

Comment:
I don't see how this could work, given that DMU_OT_NUMTYPES is 54, which is less than DMU_OT_BYTESWAP_MASK (31).  I would actually be OK in principle with adding a bit for ENCRYPTED, but it seems that there just isn't space.